fastapi/routing.py

```diff
@@ -48,6 +48,26 @@
     from pydantic.fields import Field as ModelField  # type: ignore
 
 
+def _prepare_response_content(res: Any, *, exclude_unset: bool) -> Any:
+    if isinstance(res, list):
+        return [
+            _prepare_response_content(item, exclude_unset=exclude_unset) for item in res
+        ]
+    elif isinstance(res, dict):
+        return {
+            k: _prepare_response_content(v, exclude_unset=exclude_unset)
+            for k, v in res.items()
+        }
+    elif isinstance(res, BaseModel):
```

Comment:
Is there any slight performance benefit of making this the first if condition?
Since this is likely (?) going to be the most common case.

Reply:
I actually thought about this right after pushing the PR, pushed some changes that make this the first condition (kept list as second, since I believe that is quite likely to be the second most common option). There's still the unnecessary recursive iteration for lists and dicts that do not contain any models at all, but it's hard to know if there are models without going through everything...